src/lib/ptree/json.lua

```diff
@@ -135,18 +208,50 @@ local function write_json_string(output, str)
    output:write_chars('"')
 end
 
-function write_json_object(output, obj)
+write_json_type = function(output, var)
+   local tp = type(var)
+   if tp == 'table' then
+      if var == NIL then
```

Comment:
You can hoist the NIL check outside the outer if. In my opinion, this is clearer as it's doesn't require to know NIL is encoded as a table.

```lua
function write_json_type (output, var)
   if var == NIL then write_json_null(ouput) end
   local tp = type(var)
   if tp == 'table' then
      ....
   elseif tp == 'string' then
     ....
end
```